languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml

```diff
@@ -20273,7 +20273,66 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
+        <rulegroup id="DOUBLE_INFINITIVE" name="Infinitivo duplo">
+            <short>Possível erro de digitação.</short>
+            <!-- This contains the original structure as described in premium issue 5538
+                 All sentences herein are of the shape: 'dever ter feito', and as such will usually indicate
+                 some kind of past tense epistemic modality, for which the double infinitive is very probably ungrammatical. -->
+            <rule default="temp_off">
+                <!-- This antipattern refers to constructions of the type 'the fact that...', for which
+                     a personal (i.e. inflected) infinitive can be used. -->
+                <antipattern>
+                    <token regexp="yes">fac?to|questão|possibilidade|hipótese|apesar</token>
+                    <token regexp="yes" skip="-1" postag_regexp="yes" postag="SPS00(:.+)?">
+                        d([eoa]|es[st][ea]|aquel[ea]|uma?|is[st]o)
+                        <exception postag_regexp="yes" postag="V.+"/>
+                    </token>
+                    <token postag_regexp="yes" postag="VMN.+">dever</token>
+                    <token postag_regexp="yes" postag="R." min="0"/>
+                    <token>ter</token>
+                    <token postag_regexp="yes" postag="V.P.+"/>
+                    <example>A questão de o segredo dever ter sido revelado não nos afeta.</example>
+                    <example>O fato do contrato já dever ter sido assinado é irrelevante.</example>
+                </antipattern>
+                <antipattern>
+                    <token postag="CS"/>
+                    <token postag="CS" skip="-1">
+                        de
+                        <exception postag_regexp="yes" postag="V.+"/>
+                    </token>
+                    <token postag_regexp="yes" postag="VMN.+">dever</token>
+                    <token postag_regexp="yes" postag="R." min="0"/>
+                    <token>ter</token>
+                    <token postag_regexp="yes" postag="V.P.+"/>
+                    <example>Apesar de ele dever já ter entrado, tranquei a porta.</example>
+                </antipattern>
+                <!-- A pesky little case, where 'a' + inf. actually generally fits..? -->
+                <antipattern>
+                    <token postag="SPS00">a</token>
+                    <token postag_regexp="yes" postag="VMN.+">dever</token>
+                    <token postag_regexp="yes" postag="R." min="0"/>
+                    <token>ter</token>
+                    <token postag_regexp="yes" postag="V.P.+"/>
+                    <example>A cidade ficou-lhe a dever não ter sido invadida.</example>
+                </antipattern>
+                <pattern>
+                    <token postag_regexp="yes" postag="SENT_START|[DNPARZ].+" min="1"/>
+                    <marker>
+                        <token postag_regexp="yes" postag="VMN.+">dever</token>
```

Comment:
Can we add more verbs? I can think of other fitting examples: `dever|poder|querer|pretender`

Note that `VMN.+` will also trigger a mistake like `deverem ter chegado lá`, which would likely need a different correction. `VMN0000` might be more fitting (but we could also leave it like this at first and see what kind of results we get in the diff, then add changes later)

Reply:
I started with `dever|poder` and there were some instances of 'poder' that just didn't quite fit the structures I had designed for epistemic modals. (I mean, 'poder' is also an epistemic modal often, but more frequently it's simply a potential mood marker...)

As such I thought it was a better idea to start small with `dever` only, as in the original issue, and then gradually expand to other verbs.

Reply:
We could leave it as it is and try to expand the rule later. `poder` would require some antipatterns or another opening token (`negate` + `SPS.+`, for instance).